### PR TITLE
Code Format using clang-format's Google style defaults.

### PR DESCRIPTION
**Pull request by DellaBitta**
Updated codebase using clang-format's Google style basis without any further one-off customizations.

Clang-format seemingly. does some odd things to objective-c files.  Therefore directories known to contain objective-c files are ignored by defining a local .clang-format file which disables clang-format. Objective C .mm files are ignored in mixed-language directories, and Obj-C headers are detected by the format_code.py script which search for Obj-c language tags.

### CODE REVIEW

File: app/src/locale.cc [hunk 1]
```diff
@@ -59,13 +59,11 @@ std::string GetLocale() {
...
   // default locale (empty in real life, or placeholder when running in a unit
   // test, as the test environment has no locale variables set).
-  std::string output = std::locale().name() != "C"
-                           ? std::locale().name()
-                           : getenv("LANG")
-                                 ? getenv("LANG")
-                                 : getenv("LC_CTYPE")
-                                       ? getenv("LC_CTYPE")
-                                       : getenv("TEST_TMPDIR") ? "en_US" : "";
+  std::string output = std::locale().name() != "C" ? std::locale().name()
```
Comment (alexames) [inline app/src/locale.cc:62]: nit: This is a case where I think the formatter is going to do worse than the nicely formatted table no matter what settings we use. If we go through with committing with these settings, can this block get wrapped in `// clang-format off/on`?

File: app/tests/optional_test.cc [hunk 2]
```diff
@@ -168,10 +164,7 @@ TEST_F(OptionalTest, CopyConstructor) {
   EXPECT_TRUE(another_copy_of_optional_int.has_value());
   EXPECT_EQ(another_copy_of_optional_int.value(), 9999);
 
-  SetupExpectCall()
-    .Construct(1)
-    .CopyAndMove(2, 1)
-    .Destruct(4);
+  SetupExpectCall().Construct(1).CopyAndMove(2, 1).Destruct(4);
```
Comment (alexames) [inline app/tests/optional_test.cc:167]: As mentioned above, I think this is a case where having the manual formatting is arguably superior (though I feel less strongly about this one). Another // clang-format off/on?

**Review by alexames — COMMENTED** [review]
Overall I'm much happier with how this formatting pass went, though I think there are a small handful of places we ought to throw in some // clang-format off/on statements on custom-formatted blocks.

File: app/tests/optional_test.cc (hunk 2 above)
Comment (DellaBitta) [reply app/tests/optional_test.cc]: Done.

Also found a few more.

File: app/src/locale.cc (hunk 1 above)
Comment (DellaBitta) [reply app/src/locale.cc]: Done.

**Comment by DellaBitta** [comment]
Added so clang-format on / off toggles around code which flows better in the authored state.
Also removed the firestore/.clang-format file that I inherited from their repo as it seemed to cause more changes than our standard baseline .clang-format file.

File: app/src/log_stdio.cc [hunk 3]
```diff
@@ -48,7 +48,7 @@ static const char* kLogLevelPrefix[] = {
 #if FIREBASE_PLATFORM_WINDOWS
 // Guards the log buffer on Windows.
 static Mutex g_log_mutex;  // NOLINT
-#endif  // FIREBASE_PLATFORM_WINDOWS
+#endif                     // FIREBASE_PLATFORM_WINDOWS
```
Comment (alexames) [inline app/src/log_stdio.cc:51]: (This is the kind of thing I was talking about above)

**Review by alexames — APPROVED** [review]
This looks good to me. The one minor nitpick I have is that I'm not a huge fan of when the formatter decides to line up comments at the end of lines, because it can end up with some awkwardly formatted comments, but I don't think it's worth overriding google style to turn that off.

**Comment by DellaBitta** [comment]
> This looks good to me. The one minor nitpick I have is that I'm not a huge fan of when the formatter decides to line up comments at the end of lines, because it can end up with some awkwardly formatted comments, but I don't think it's worth overriding google style to turn that off.

Yeah, I saw this too and it made me sigh.

**Pull request merged**